### PR TITLE
Make size calculation progressive

### LINKED ISSUE #1043: High cpu and ram usage when calculating the size of folders containing a large number of files
**UPDATE:** After more tests, it only happened the first time, then the use of cpu and ram is not that high anymore, but it is still a bit high, and it takes too long to show the size.
I also realized that it is not so much for the size, but for the amount of file.

while doing the calculation:
Captura de pantalla (22)

after finishing the calculation (after a few seconds it returns to normal):
Captura de pantalla (23)

### PR DESCRIPTION
**Pull request by hez2010**
- Added ability to cancel size calculation
- Make size calculation progressive

Closes #1043

### CODE REVIEW

**Comment by tsvietOK** [comment]
Why this PR contains changes from your another PR?

**Comment by hez2010** [comment]
@tsvietOK I've rebased my branch.

File: Files/Views/Pages/Properties.xaml [hunk 1]
```diff
@@ -143,7 +143,15 @@
                     Grid.Column="1"
                     Margin="20,0,0,0"
                     IsTextSelectionEnabled="True"
-                    Text="{x:Bind ItemProperties.ItemSize, Mode=OneWay}" />
+                    Text="{x:Bind ItemProperties.ItemSize, Mode=OneWay}"
+                    Visibility="{x:Bind ItemProperties.ItemSizeVisibility, Mode=OneWay}" />
+                <ProgressBar
```
Comment (tsvietOK) [inline Files/Views/Pages/Properties.xaml:148]: Add namespace `xmlns:muxc="using:Microsoft.UI.Xaml.Controls"` to the root element of this document and change all ProgressBars in this document to `muxc:ProgressBar`. In `GetHashForFile` method change `ProgressBar` to `Microsoft.UI.Xaml.Controls.ProgressBar`.

File: Files/Views/Pages/Properties.xaml.cs [hunk 2]
```diff
@@ -146,13 +150,39 @@ private void Properties_Unloaded(object sender, RoutedEventArgs e)
...
+                long size = 0;
+                uint index = 0;
+                try
+                {
+                    do
+                    {
+                        var files = await folders.GetFilesAsync(index, 16);
+                        count = files.Count;
+                        index += (uint)count;
+                        size += (await Task.WhenAll(files.Select(async file => (await file.GetBasicPropertiesAsync()).Size)))
```
Comment (tsvietOK) [inline Files/Views/Pages/Properties.xaml.cs:168]: Is it possible to display current size value while calculating is not ended?

Comment (hez2010) [reply Files/Views/Pages/Properties.xaml.cs:168]: Done

File: Files/Views/Pages/Properties.xaml (hunk 1 above)
Comment (hez2010) [reply Files/Views/Pages/Properties.xaml]: Done

**Comment by hez2010** [comment]
Closing this PR since it's covered by #1099.

**Pull request closed without merging**